advisor-backend/routes/schedule.js

```diff
@@ -208,6 +292,47 @@ const generateSchedule = async (
   return years;
 };
 
+const moveCourseToDifferentYear = (
+  schedule,
+  desiredYear,
+  originalYear,
+  course
+) => {
+  for (let i = 0; i < schedule.length; i++) {
+    const year = schedule[i];
+    // delete it from this year
+    if (year.number === originalYear) {
+      // for both semesters
+      year.semesters[0].classes = year.semesters[0].classes.filter(
+        (classItem) => classItem.id !== course.id
+      );
+      year.semesters[1].classes = year.semesters[0].classes.filter(
+        (classItem) => classItem.id !== course.id
+      );
+    }
+
+    // add it to this year
+    if (year.number === desiredYear) {
```

Comment:
Do classes always happen across the full year? With the logic as is now you might only filter it from one semester, but then add it to both.